packages/syft/proto/core/tensor/share_tensor.proto

```diff
@@ -0,0 +1,15 @@
+syntax = "proto3";
+
+package syft.core.tensor;
+
+import "proto/core/tensor/tensor.proto";
+import "proto/lib/numpy/array.proto";
+
+message ShareTensor {
+    oneof data {
+        Tensor tensor = 1;
```

Comment:
Is there a reason there is both "tensor" and "array"?

Reply:
I think this was from a branch where I had also the DP code. I will revisit this logic

Reply:
Ahhh..it is because the child could be a Tensor or an array -- I am not sure which approach should we keep (I think this was something that I have seen in the DP branch and I carried with it)
If you look here - basically there is an `if...else` depending on the child [here](https://github.com/OpenMined/PySyft/pull/5822/files#diff-37698445e24479cdf1304dfe050c1beb3d4e6463eca4245b8a52591b1d8028fd) at the serialization

Reply:
Ah right - that makes sense

Reply:
Yeah custom tensor proto is recursive and can be another Tensor all the way to the bottom until it finally must be a numpy array. Turtles most of the way down. 🐢 🌏

Reply:
Having numpy also , includes unsigned integers(not available in torch), which helped me a lot  in protocol implementations.